src/features/web-compat.js

```diff
@@ -512,15 +512,46 @@ export default class WebCompat extends ContentFeature {
     }
 
     viewportWidthFix () {
-        const viewportTag = document.querySelector('meta[name=viewport]')
-        if (!viewportTag) return
-        const viewportContent = viewportTag.getAttribute('content')
-        if (!viewportContent) return
-        const viewportContentParts = viewportContent.split(',')
-        const widthPart = viewportContentParts.find((part) => part.includes('width'))
-        // If we already have a width, don't add one
-        if (widthPart) return
-        viewportTag.setAttribute('content', `${viewportContent},width=device-width`)
+        const viewportTags = document.querySelectorAll('meta[name=viewport]')
+        // Chrome respects only the last viewport tag
+        let viewportTag = viewportTags.length === 0 ? null : viewportTags[viewportTags.length - 1]
+        const viewportContent = viewportTag?.getAttribute('content')
+        const viewportContentParts = viewportContent?.split(/,|;/)
+        if (!viewportTag || this.desktopModeEnabled) {
+            // force wide viewport width
+            const viewportTagExists = Boolean(viewportTag)
+            if (!viewportTag) {
+                viewportTag = document.createElement('meta')
+                viewportTag.setAttribute('name', 'viewport')
+            }
+            const forcedWidth = screen.width >= 1280 ? 1280 : 980
+            // Race condition: depending on the loading state of the page, initial scale may or may not be respected, so the page may look zoomed-in after applying this hack.
+            // Usually this is just an annoyance, but it may be a bigger issue if user-scalable=no is set, so we remove it too.
+            const forcedInitialScale = (screen.width / forcedWidth).toFixed(3)
+            const newContentParts = [`width=${forcedWidth},initial-scale=${forcedInitialScale}`]
+            viewportContentParts?.forEach((part) => {
+                const key = part.split('=')[0].trim().toLowerCase()
+                if (!['width', 'initial-scale', 'user-scalable'].includes(key)) {
+                    newContentParts.push(part)
+                }
+            })
+            viewportTag.setAttribute('content', newContentParts.join(','))
+            if (!viewportTagExists) {
+                document.head.appendChild(viewportTag)
+            }
+        } else { // mobile mode with a viewport tag
+            // fix an edge case where WebView forces the wide viewport
+            const widthPart = viewportContentParts?.find((part) => part.includes('width'))
```

Comment:
Should we lower and trim these too? Sorry for follow up here.

Reply:
thanks for catching this, it slipped my attention!